File: backend/app/schemas/team.py
from pydantic import BaseModel
from typing import Optional, List

class TeamBase(BaseModel):
    team_name: str
    external_id: Optional[str] = None

class TeamCreate(TeamBase):
    team_name: str

class TeamUpdate(BaseModel):
    team_name: Optional[str] = None
    external_id: Optional[str] = None

class TeamResponse(TeamBase):
    id: str
    tournament_names: Optional[List[str]] = None  # Tournaments the team has played in
    
    class Config:
        from_attributes = True

File: frontend/src/types/team.ts
export interface Team {
  id: string;
  team_name: string;
  external_id?: string;
  tournament_names?: string[]; // Tournaments the team has played in
}

export interface TeamCreate {
  team_name: string;
  external_id?: string;
}

export interface TeamUpdate {
  team_name?: string;
  external_id?: string;
}
